skimage/exposure/exposure.py

```diff
@@ -368,14 +414,24 @@ def equalize_hist(image, nbins=256, mask=None):
 
     Notes
     -----
-    This function is adapted from [1]_ with the author's permission.
+    The `method='float'` implementation is adapted from [1]_ with the author's
+    permission. The `method='uint8'` implementation follows a standard
+    approach as described in [2]_.
 
     References
     ----------
     .. [1] http://www.janeriksolem.net/histogram-equalization-with-python-and.html
-    .. [2] https://en.wikipedia.org/wiki/Histogram_equalization
+    .. [2] https://en.wikipedia.org/wiki/Histogram_equalization#Examples
 
     """
+    if method == 'uint8':
+        if image.dtype != np.uint8:
+            raise ValueError(
+                "method 'uint8' requires image to have np.uint8 dtype"
+            )
+        return _equalize_hist_uint8(image, mask)
+    elif method != 'float':
+        raise ValueError("method must be 'uint8' or 'float'")
     if mask is not None:
         mask = np.array(mask, dtype=bool)
```

Comment:
This conversion is required before the method selection (`uint8` or `float`)

Reply:
right, nice catch